File: TESTING/LIN/derrorhr_col.f
      SUBROUTINE DERRORHR_COL( PATH, NUNIT )
      IMPLICIT NONE
*
*  -- LAPACK test routine --
*  -- LAPACK is a software package provided by Univ. of Tennessee,    --
*  -- Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd..--
*
*     .. Scalar Arguments ..
      String   (LEN=3)   PATH;
      int                NUNIT
*     ..
*
*  =====================================================================
*
*     .. Parameters ..
      int                NMAX
      PARAMETER          ( NMAX = 2 )
*     ..
*     .. Local Scalars ..
      int                I, INFO, J
*     ..
*     .. Local Arrays ..
      double             A( NMAX, NMAX ), T( NMAX, NMAX ), D(NMAX);
*     ..
*     .. External Subroutines ..
      EXTERNAL           ALAESM, CHKXER, DORHR_COL
*     ..
*     .. Scalars in Common ..
      bool               LERR, OK;
      String   (LEN=32)  SRNAMT;
      int                INFOT, NOUT
*     ..
*     .. Common blocks ..
      COMMON             / INFOC / INFOT, NOUT, OK, LERR
      COMMON             / SRNAMC / SRNAMT
*     ..
*     .. Intrinsic Functions ..
      INTRINSIC          DBLE
*     ..
*     .. Executable Statements ..
*
      NOUT = NUNIT
      WRITE( NOUT, FMT = * )
*
*     Set the variables to innocuous values.
*
      DO J = 1, NMAX
         DO I = 1, NMAX
            A( I, J ) = 1.D+0 / DBLE( I+J )
            T( I, J ) = 1.D+0 / DBLE( I+J )
         END DO
         D( J ) = 0.D+0
      END DO
      OK = .TRUE.
*
*     Error exits for Householder reconstruction
*
*     DORHR_COL
*
      SRNAMT = 'DORHR_COL'
*
      INFOT = 1
      CALL DORHR_COL( -1, 0, 1, A, 1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      INFOT = 2
      CALL DORHR_COL( 0, -1, 1, A, 1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
      CALL DORHR_COL( 1, 2, 1, A, 1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      INFOT = 3
      CALL DORHR_COL( 0, 0, -1, A, 1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      CALL DORHR_COL( 0, 0, 0, A, 1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      INFOT = 5
      CALL DORHR_COL( 0, 0, 1, A, -1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      CALL DORHR_COL( 0, 0, 1, A, 0, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      CALL DORHR_COL( 2, 0, 1, A, 1, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      INFOT = 7
      CALL DORHR_COL( 0, 0, 1, A, 1, T, -1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      CALL DORHR_COL( 0, 0, 1, A, 1, T, 0, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
      CALL DORHR_COL( 4, 3, 2, A, 4, T, 1, D, INFO )
      CALL CHKXER( 'DORHR_COL', INFOT, NOUT, LERR, OK )
*
*     Print a summary line.
*
      CALL ALAESM( PATH, OK, NOUT )
*
      RETURN
*
*     End of DERRORHR_COL
*
      END

File: test/lin/derrorhr_col.dart
// Copyright (c) 2024 Guilherme Lepsch. All rights reserved. Use of this
// source code is governed by a BSD-style license that can be found in the
// [LICENSE file](https://github.com/lepsch/dart_lapack/blob/main/LICENSE).

import 'package:dart_lapack/lapack.dart';

import '../test_driver.dart';
import 'alaesm.dart';
import 'chkxer.dart';
import 'common.dart';

void derrorhr_col(final String PATH, final Nout NUNIT, final TestDriver test) {
// -- LAPACK test routine --
// -- LAPACK is a software package provided by Univ. of Tennessee,    --
// -- Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd..--
  const NMAX = 2;
  final A = Matrix<double>(NMAX, NMAX),
      T = Matrix<double>(NMAX, NMAX),
      D = Array<double>(NMAX);
  final INFO = Box(0);

  final NOUT = infoc.NOUT = NUNIT;
  final OK = infoc.OK;
  final LERR = infoc.LERR;
  NOUT.println();

  // Set the variables to innocuous values.

  for (var J = 1; J <= NMAX; J++) {
    for (var I = 1; I <= NMAX; I++) {
      A[I][J] = 1.0 / (I + J);
      T[I][J] = 1.0 / (I + J);
    }
    D[J] = 0.0;
  }
  OK.value = true;

  // Error exits for Householder reconstruction

  test('DORHR_COL', () {
    srnamc.SRNAMT = 'DORHR_COL';

    infoc.INFOT = 1;
    dorhr_col(-1, 0, 1, A, 1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    infoc.INFOT = 2;
    dorhr_col(0, -1, 1, A, 1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);
    dorhr_col(1, 2, 1, A, 1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    infoc.INFOT = 3;
    dorhr_col(0, 0, -1, A, 1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    dorhr_col(0, 0, 0, A, 1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    infoc.INFOT = 5;
    dorhr_col(0, 0, 1, A, -1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    dorhr_col(0, 0, 1, A, 0, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    dorhr_col(2, 0, 1, A, 1, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    infoc.INFOT = 7;
    dorhr_col(0, 0, 1, A, 1, T, -1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    dorhr_col(0, 0, 1, A, 1, T, 0, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);

    dorhr_col(4, 3, 2, A, 4, T, 1, D, INFO);
    chkxer('DORHR_COL', infoc.INFOT, NOUT, LERR, OK, test);
  });

  // Print a summary line.
  alaesm(PATH, OK.value, NOUT);
}
